src/matcha_ml/cli/destroy.py

```diff
@@ -27,23 +27,25 @@ def destroy_resources() -> None:
         raise typer.Exit()
 
     with remote_state.use_lock():
-        # create a runner for deprovisioning resource with Terraform service.
-        template_runner = AzureTemplateRunner()
-
-        if not check_current_deployment_exists():
-            print_error(
-                "Error - you cannot destroy resources that have not been provisioned yet."
-            )
-            raise typer.Exit()
-
-        if template_runner.is_approved(verb="destroy"):
-            # deprovision the resources
-            template_runner.deprovision()
-            print_status(build_step_success_status("Destroying resources is complete!"))
-        else:
-            print_status(
-                build_status(
-                    "You decided to cancel - your resources will remain active! If you change your mind, then run 'matcha destroy' again."
+
+        with remote_state.use_remote_state():
+            # create a runner for deprovisioning resource with Terraform service.
+            template_runner = AzureTemplateRunner()
```

Comment:
can we move both this line and `if not check_current` check out of context as we don't need a context for that?

Reply:
I thought we would need the context for that, as we'd need to download the remote state first to check against?